include/kangaru.hpp

```diff
@@ -198,12 +200,19 @@ class Container : public std::enable_shared_from_this<Container> { public:
 	constexpr static detail::enabler null = {};
 	
 public:
-	Container() = default;
-	Container(const Container &) = default;
-	Container(Container &&) = default;
-	Container& operator =(const Container &) = default;
-	Container& operator =(Container &&) = default;
-	~Container() = default;
+	Container(const Container &) = delete;
+	Container(Container &&) = delete;
+	Container& operator =(const Container &) = delete;
+	Container& operator =(Container &&) = delete;
+	virtual ~Container() = default;
+
+	template <typename T, typename... Args, 
+		 enable_if<is_base_of_container<T>> = null>
+	static std::shared_ptr<T> make_container(Args&&... args) {
```

Comment:
Since `init()` function moved to `protected` section, we can call it only inside our class functions.